rl4co/envs/common/base.py

```diff
@@ -127,6 +127,16 @@ def step(self, td: TensorDict) -> TensorDict:
             # Since we simplify the syntax
             return self._torchrl_step(td)
 
+    def reset(self, td: Optional[TensorDict] = None, batch_size=None) -> TensorDict:
+        """Reset function to call at the beginning of each episode"""
+        if batch_size is None:
+            batch_size = self.batch_size if td is None else td.batch_size
+        if td is None or td.is_empty():
+            td = self.generator(batch_size=batch_size)
```

Comment:
Yes the reason why they are here is because they could get passed to the environment itself, as in TorchRL [here](https://github.com/pytorch/rl/blob/583e2a1657f8b67428560c0dfeb0130d7240a0dd/torchrl/envs/common.py#L343). This is the function signature for `EnvBase` in TorchRL:

```python
    def __init__(
        self,
        *,
        device: DEVICE_TYPING = None,
        batch_size: Optional[torch.Size] = None,
        run_type_checks: bool = False,
        allow_done_after_reset: bool = False,
    ):
    ```
    So I guess we should make the above explicit in `RL4COEnvBase` since ours is a child class!

Reply:
@ngastzepeda I think it makes sense that now all environments have the `generator` and `generator_params` as the input in the `__init__()` function. We could move them to the `RL4COEnvBase()`. 

And also as @fedebotu said, it's better to show other useful parameters from `torchrl.EnvBase` in our `RL4COEnvBase()`. Easier for users to know the provided APIs not only from the documentation.

Reply:
@ngastzepeda I rethink about adding `generator` and `generator_params` to the `RL4COEnvBase` class. I prefer to keep them for each environment at the current moment for two reasons:

1. We want user can initialize the environment with simple calling `env = <EnvName>()`, e.g. `env = TSPEnv()` without any parameters. In this case, there is a required generator initialization for each environment with respective generator class, e.g. https://github.com/ai4co/rl4co/blob/a9943c936af1bd3803d377ba0c39b49250e98101/rl4co/envs/routing/cvrp/env.py#L59-L61
It would be hard or at least massive for users to understand if we implement this part in the base class;

2. Putting the generator initializing for each environment could be a hint for users to understand the "generate data" -> "reset instance as a tensordict" -> "step rollout, ..." pipeline. 

What do you think about this? 🤔